protocol.txt

```diff
@@ -0,0 +1,124 @@
+Starting from ESP firmware version 10, we are using new wire protocol.
+All messages have a fix-sized header and a variable-sized payload.
+Generic message description:
+
+        0               1               2               3
+        0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7 0 1 2 3 4 5 6 7
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     0 |                                                               |
+       +                            Intron                             +
+     8 |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    12 |     Type      |     Byte      |             Size              |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    16 |                                                               |
+       ~                                                               ~
+       ~                            Payload                            ~
+       ~                                                               ~
+       |                                                               |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+    Intron:         Each message is prefixed by 8 bytes. This is used
+                    to ensure synchronization, because we are using
+                    unreliable UART.
+                    Starting value is 'UN\x00\x01\x02\x03\x04\x05'.
+                    Printer may change it, see MSG_CLIENTCONFIG_V2.
+
+    Type:           This byte indicates type of the message.
+                    Possible values are:
+                        0: MSG_DEVINFO_V2
+                        6: MSG_CLIENTCONFIG_V2
+                        7: MSG_PACKET_V2
+                    Values 1-5 are reserved and must not be used.
```

Comment:
Reserved usually means „for future use“. Should this be „discontinued“ or „deprecated“?